libsolidity/analysis/DeclarationTypeChecker.cpp

```diff
@@ -140,6 +214,9 @@ void DeclarationTypeChecker::endVisit(Mapping const& _mapping)
 
 void DeclarationTypeChecker::endVisit(ArrayTypeName const& _typeName)
 {
+	if (_typeName.annotation().type)
```

Comment:
All the ifs of this form, is it expected that they'll be true/false somehow?

Reply:
Some visits can happen multiple times for example during the recursive struct resolution. I'm not sure it's worth trying to avoid that entirely.

Reply:
OK!